homeassistant/components/vacuum/roomba.py

```diff
@@ -0,0 +1,264 @@
+        self._is_on = True
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Turn the vacuum off and return to home."""
+        yield from self.async_stop()
+        yield from self.async_return_to_base()
+
+    @asyncio.coroutine
+    def async_stop(self, **kwargs):
+        """Stop the vacuum cleaner."""
+        yield from self.hass.async_add_job(self.vacuum.send_command, 'stop')
+        self._is_on = False
+
+    @asyncio.coroutine
+    def async_resume(self, **kwargs):
+        """Resume the cleaning cycle."""
+        yield from self.hass.async_add_job(self.vacuum.send_command, 'resume')
+        self._is_on = True
+
+    @asyncio.coroutine
+    def async_pause(self, **kwargs):
+        """Pause the cleaning cycle."""
+        yield from self.hass.async_add_job(self.vacuum.send_command, 'pause')
+        self._is_on = False
+
+    @asyncio.coroutine
+    def async_start_pause(self, **kwargs):
+        """Pause the cleaning task or resume it."""
+        if self.vacuum_state and self.is_on:  # vacuum is running
+            yield from self.async_pause()
+        elif self._status == 'Stopped':  # vacuum is stopped
+            yield from self.async_resume()
+        else:  # vacuum is off
+            yield from self.async_turn_on()
+
+    @asyncio.coroutine
+    def async_return_to_base(self, **kwargs):
+        """Set the vacuum cleaner to return to the dock."""
+        yield from self.hass.async_add_job(self.vacuum.send_command, 'dock')
+        self._is_on = False
+
+    @asyncio.coroutine
+    def async_send_command(self, command, params, **kwargs):
+        """Send raw command."""
+        _LOGGER.debug("async_send_command %s (%s), %s",
+                      command, params, kwargs)
+        yield from self.hass.async_add_job(
+            self.vacuum.send_command, command, params)
+        return True
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Fetch state from the device."""
+        # No data, no update
+        if not self.vacuum.master_state:
+            _LOGGER.debug("Roomba %s has no data yet. Skip update.",
+                          self.name)
+            return
+        state = self.vacuum.master_state.get('state', {}).get('reported', {})
+        _LOGGER.debug("Got new state from the vacuum: %s", state)
+        self.vacuum_state = state
+        self._available = state is not None
+
+        # Get the capabilities of our unit
+        capabilities = state.get('cap', {})
+        cap_pos = capabilities.get('pose', None)
+        cap_bin_full = capabilities.get('binFullDetect', None)
+
+        bin_state = state.get('bin', {})
+
+        # Roomba software version
+        software_version = state.get('softwareVer', None)
+
+        # Error message in plain english
+        error_msg = self.vacuum.error_message
+
+        self._battery_level = state.get('batPct', None)
+        self._status = self.vacuum.current_state
+        self._is_on = self._status in ['Running']
```

Comment:
I suspected there were more than one state in which case the vac should be considered "on". But since I could not find any, yeah this does not make sense anymore.